v0/destinations/salesforce/transform.js

```diff
@@ -235,11 +235,43 @@ async function getSalesforceIdFromPayload(message, authorizationData) {
         leadObjectId = leadQueryResponse.data.searchRecords[0].Id;
       }
     }
-
-    // add a Lead Object to the response
-    salesforceMaps.push({ salesforceType: "Lead", salesforceId: leadObjectId });
+    // if leadObjectId is undefined => push it into the SalesforceMap. SF will create a leadObjectId
+    // for it, which can be referenced later to upfate, or check for converted status.
+    // -----------------------
+    // Checking lead for converted status, whether it is lead or contact.
+    try {
+      if (leadObjectId) {
+        const convertedDetails = await axios.get(
+          `${authorizationData.instanceUrl}/services/data/v${SF_API_VERSION}/sobjects/Lead/${leadObjectId}?fields=IsConverted,ConvertedContactId,IsDeleted`,
```

Comment:
Can we merge this call with the call above on line #211?
Something like `${authorizationData.instanceUrl}/services/data/v${SF_API_VERSION}/sobjects/Lead/${leadObjectId}?fields=id,IsConverted,ConvertedContactId,IsDeleted`

That way we have all the necessary information and don't have to make another call? I was able to confirm that this works for converted leads but I did not have a chance to confirm for non converted leads. I just know that Salesforce has strict API limits so if this does count against the limit, the less calls the better

Reply:
We had tried this, but the call was failing so we fetched the `IsConverted`, `ConvertedContactId` and `IsDeleted` after querying them with leadObjectId here rather than email.

Reply:
Resolving this comment to unblock the release.